### PR TITLE
Cumulus 470 Lambda to Update S3 bucket policy

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Lambda and tests for updating bucket policy

Addresses [CUMULUS-470](https://bugs.earthdata.nasa.gov/browse/CUMULUS-470)

## Changes

* Lambda sets S3 bucket policy
* Tests that create a new bucket, set the policy, test access from east and west
* Fixed IAM policies with incorrect resources

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing
- [x] Update CHANGELOG
- [x] Run `./bin/eslint-ratchet` and verify that eslint errors have not increased
  - [x] Commit `.eslint-ratchet-high-water-mark` if the score has improved

## Release PR

If this is a **release** PR, make sure you branch name starts with `release-` prefix, e.g. `release-v-1.1.2`.

Reviewers: @scisco @sethvincent @kkelly51 @jennyhliu

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -115,8 +115,11 @@ jobs:
               cd example
               sed -i -- 's/npm/src/g' spec/config.yml
 
+              # deploy latest iam roles to aws
+              ./node_modules/.bin/kes cf deploy --kes-folder iam --region us-east-1 --deployment cumulus-from-source
```
Comment (scisco) [inline .circleci/config.yml]: I say this is redundant. In the unlikely case of updating the iam, we just have to update it from a local machine.

File: .circleci/config.yml [hunk 2]
```diff
@@ -170,6 +173,7 @@ jobs:
           name: Deploy cumulus with the npm packages source code
           command: |
             cd example
+            ./node_modules/.bin/kes cf deploy --kes-folder iam --region us-east-1 --deployment cumulus-from-npm
```
Comment (scisco) [inline .circleci/config.yml]: same comment here

**Comment by scisco** [comment]
@laurenfrederick I added a few changes to your PR to make the integration tests faster and more stable.

File: .circleci/config.yml (hunk 1 above)
Comment (laurenfrederick) [reply .circleci/config.yml]: So the thing is, these tests will not work without updating the iam, which is why I put it in. So everyone will have to update their iams after I merge. Also I'd have to update the npm test stack (source was updated with this). Is it fine to leave it? Just seems easier to automatically grab any iam changes.

Comment (scisco) [reply .circleci/config.yml]: sounds good. let's keep it here.

**Pull request merged**